### DISCUSSION TITLE
Rolling up in bulk on demand, but not from flow

### DISCUSSION THREAD

**Discussion by dafyddjohns** [jamessimone/apex-rollup#342] [Q&A]
I was recently introduced to this amazing package after I fell into the platform limitation around roll up summaries on currency fields from Opportunity to Account when Advanced Currency Management is switched on.

I've followed the documentation as best as I can, and have got the stage where running retrospective jobs on existing data produces the expected output. 

However when I try to invoke the one and only CMDT record I have configured from an after save flow, I am not getting any Queuable instances created, even though I can see that the record ID of the Opp does get passed in. I have tried setting the defer parameter to false, and tried with defer = true with the subsequent call to execute deferred actions, but in both instances, nothing is happening.
<img width="450" alt="Screenshot 2022-06-30 at 22 20 42" src="https://user-images.githubusercontent.com/37784457/176780267-637d71cc-aa7a-43b7-8fb0-7b71604e4318.png">

Is there anything I'm missing?

<img width="1004" alt="Screenshot 2022-06-30 at 22 18 48" src="https://user-images.githubusercontent.com/37784457/176780058-f6d7c7b5-661a-4ac6-8f9a-cf5d8a5fe0e7.png">
<img width="832" alt="Screenshot 2022-06-30 at 22 19 28" src="https://user-images.githubusercontent.com/37784457/176780135-02c79626-f20b-4bcf-9c28-e67d949ec192.png">
<img width="524" alt="Screenshot 2022-06-30 at 22 19 34" src="https://user-images.githubusercontent.com/37784457/176780147-03b283fb-77fc-46e0-b043-bb93a296c2df.png">

**Comment by jamessimone** [comment]
@dafyddjohns sorry for the delay in response - is your org default for the Rollup Setting enabled? That tends to be the issue for the first action. If you _do_ end up not setting defer, or setting it to true, you’ll also need the “process deferred rollups” action!

hopefully that helps! Let me know, happy to assist further — and welcome to the Apex Rollup world!

**Reply by dafyddjohns** [reply]
Sorry I should've got a bit more details up.

Yes I've created the org default and ensured that it's enabled there. The invocable action definitely gets called from the flow too, which I can see in debug logs and the flow debugger. The record that I've filtered out do get passed to the invocable method too, both the new and old.

I've kept the defer to the default of false, and it didn't trigger anything, so I also tried defer true with the subsequent call to process defferred rollups but that also resulted in the parent not being updated and no Queueables created.

A link to a debug log I generated : 
[apex-07L5t000003NvVtEAK (1).log](https://github.com/jamessimone/apex-rollup/files/9027089/apex-07L5t000003NvVtEAK.1.log)

Thanks for the support!

**Reply by jamessimone** [reply]
Ah you are debugging the action within Flow Builder? Are you testing with an opportunity update, by chance? Apex Rollup does a diff-based rollup for SUM, which (for updates) compares the old/new values of your rollup field and only performs a rollup if those values have changed.

**Comment by dafyddjohns** [comment]
Ah potentially not related to the observations above, but I also see under special considerations that the dated exchange rates aren't yet handled. I might have to build something quick and specific out for this use case as it's something that's cropped up during a deployment.

**Reply by jamessimone** [reply]
It’s interesting that you were seeing the correct amounts through the full recalc app; I just haven’t tested with a dated exchange rate org yet. If you’re at all familiar with the underworkings of that and Apex, I’d encourage you to check out the `RollupCurrencyInfo` class as you would probably be able to tell if there were further changes needed.
